### PR TITLE
Merging #180 @psmit hidden command impl

### PR DESCRIPTION
**Pull request by meatballhat**
Closes #180

### CODE REVIEW

File: app.go [hunk 1]
```diff
@@ -357,11 +372,41 @@ func (a *App) Command(name string) *Command {
 	return nil
 }
 
-// Returnes the array containing all the categories with the commands they contain
+// Categories returns a slice containing all the categories with the commands they contain
```
Comment (jszwedko) [inline app.go]: 👍

File: app.go [hunk 2]
```diff
@@ -357,11 +372,41 @@ func (a *App) Command(name string) *Command {
...
 func (a *App) Categories() CommandCategories {
 	return a.categories
 }
 
+// VisibleCategories returns a slice of categories and commands that are
+// Hidden=false
+func (a *App) VisibleCategories() []*CommandCategory {
+	ret := []*CommandCategory{}
+	for _, category := range a.categories {
+		if visible := func() *CommandCategory {
```
Comment (jszwedko) [inline app.go]: I have never seen this idiom before, clever!

File: help.go [hunk 3]
```diff
@@ -188,7 +192,10 @@ func printHelp(out io.Writer, templ string, data interface{}) {
 	err := t.Execute(w, data)
 	if err != nil {
 		// If the writer is closed, t.Execute will fail, and there's nothing
-		// we can do to recover. We could send this to os.Stderr if we need.
+		// we can do to recover.
+		if os.Getenv("CLI_TEMPLATE_ERROR_DEBUG") != "" {
```
Comment (jszwedko) [inline help.go]: 👏

**Comment by jszwedko** [comment]
@meatballhat can you add a spec for the `VisibleCategories` and `VisibleCommands` logic? Otherwise :shipit:

**Comment by meatballhat** [comment]
@jszwedko howzat look?

**Comment by jszwedko** [comment]
I couldn't ask for anything more.

**Comment by jszwedko** [comment]
Added to CHANGELOG in https://github.com/codegangsta/cli/commit/28eb7b2cc4bf1b4d266fe55fe7d94f958170cdf8

**Pull request merged**